### ISSUE TITLE
training on server 

### ISSUE THREAD

**Issue by cudanexus** [AlexeyAB/darknet#2674]
@AlexeyAB Hello,

**1.** I am trying to send training data means training steps, Avg IOU, Avg loss, loss, image count Values  to the server to show a training graph on the webpage. I manage to do it by saving terminal training output in a text file and extract values from it and send it to the server. Is there any other way to do that. Or Is this the correct approch.

**2.** I want to do that for -map flag, test flag. Because to I want to make it web-based UI. Any help would be great.

**Comment by AlexeyAB** [comment]
@cudanexus Hi,


1. You can try do this as you desribed. 
If it will be required, I can try to do JSON sender of Training output.
Or you can train by using
`/darknet detector train cfg/coco.data yolov3.cfg darknet53.conv.74 -map | tee -a log.txt`
and send `log.txt` to another server, or another server read this `log.txt`.
Or by using some of these approaches: https://unix.stackexchange.com/questions/151316/how-to-send-a-output-of-unix-command-from-one-server-to-another-server


2. Currently Darknet can send Loss & mAP chart by using HTTP-JPEG-stream
`/darknet detector train cfg/coco.data yolov3.cfg darknet53.conv.74 -dont_show -mjpeg_port 8090 -map`
So you can open in browser `http://192.168.0.30:8090/` and see the Loss & mAP chart in real-time
You can see it by using: Web-browser, VLC, OpenCV, ...

You can decrease this value from `500000` to `50000` if sometimes your training is going slowed due to bad connection: https://github.com/AlexeyAB/darknet/blob/1cd332e4cac001ffcc12a24c72640fe02b69a8a0/src/image.c#L832

----

**Comment by cudanexus** [comment]
If i get JSON output for training I can use it for real-time graph each and every step of training @AlexeyAB can you tell me which line and what should i write to get json output
I have changed [region_layer #L363 ](https://github.com/AlexeyAB/darknet/blob/1cd332e4cac001ffcc12a24c72640fe02b69a8a0/src/region_layer.c#L363)
So i am getting only this output in terminal 
```
208: 13.461194, 18.660343 avg loss, 0.000002 rate, 0.522565 seconds, 13312 images
Loaded: 0.000028 seconds
```
So you can tell me how to dump all this in json realtiime because tee is not working in realtime its taking around 35 steps than writing to txt file

**Comment by AlexeyAB** [comment]
> tee is not working in realtime its taking around 35 steps than writing to txt file

You can try command: https://stackoverflow.com/questions/11337041/force-line-buffering-of-stdout-when-piping-to-tee

`stdbuf -o 0 /darknet detector train cfg/coco.data yolov3.cfg darknet53.conv.74 -dont_show -mjpeg_port 8090 -map | tee output.txt`

----

To implement sending information, you can use function
`send_json_custom(char const* send_buf, int port, int timeout);`

For example, you can change this line: 
to these lines:
```cpp
        char out_buff[1024];
        sprintf(out_buff, "\n %d: %f, %f avg loss, %f rate, %lf seconds, %d images\n", get_current_batch(net), loss, avg_loss, get_current_rate(net), (what_time_is_it_now() - time), i*imgs);
        printf("%s", out_buff);
        send_json_custom(out_buff, 8070, 500000);
```
